sql/src/test/java/io/crate/operation/udf/JavascriptUserDefinedFunctionTest.java

```diff
@@ -205,17 +204,12 @@ public void testEvaluateBytesRefConvertedToString() throws Exception {
     @Test
     public void testEvaluateBytesRefInObjectIsConvertedToString() throws Exception {
         registerUserDefinedFunction("f", DataTypes.OBJECT, ImmutableList.of(DataTypes.OBJECT),
-            "function f(o) { return {'foo' :o['outer']['foo'][0], 'bar': o['outer']['bar']}; }");
-        Map<String, Object> outer = new HashMap<>();
+            "function f(o) { return {'foo' : o['inner']['foo'][0], 'bar': o['inner']['bar']}; }");
+        // the map will be modified
         Map<String, Object> inner = new HashMap<>();
         inner.put("foo", new Object[]{new BytesRef("bar")});
         inner.put("bar", new BytesRef("foo"));
-        outer.put("outer", inner);
-
-        Map<String, Object> expected = new HashMap<>();
-        expected.put("bar", "foo");
-        expected.put("foo", "bar");
-        assertEvaluate("f(obj)", expected, Literal.of(outer));
+        assertEvaluate("f(obj)", ImmutableMap.of("bar", "foo", "foo", "bar"), Literal.of(ImmutableMap.of("inner", inner)));
```

Comment:
having a map where the key->val repeats reverse is really confusing and not really good readable imho. is this really needed?